### PR TITLE
Update parity-ui-precompiled with branch

### PR DESCRIPTION
**Pull request by jacogr**
Closes https://github.com/ethcore/parity/issues/4813

Explicitly specify branch in Cargo.toml and update based on where we are. (So will reflect stable, beta or master). Saves manual work and mistakes.

### CODE REVIEW

File: js/scripts/release.sh [hunk 1]
```diff
@@ -89,11 +89,14 @@ fi
 
 echo "*** Updating cargo parity-ui-precompiled#$PRECOMPILED_HASH"
 git submodule update
+cat dapps/ui/Cargo.toml | sed "s/^parity-ui-precompiled = { git =.*/parity-ui-precompiled = { git = \"https:\/\/github.com\/ethcore\/js-precompiled.git\", optional = true, branch = \"$BRANCH\" }/" >dapps/ui/Cargo.toml.update
```
Comment (tomusdrw) [inline js/scripts/release.sh]: `sed -i dapps/ui/Cargo.toml "s/^parity-ui-precompiled = { git =.*/parity-ui-precompiled = { git = \"https:\/\/github.com\/ethcore\/js-precompiled.git\", optional = true, branch = \"$BRANCH\" }/"`
Should suffice.

Comment (jacogr) [inline js/scripts/release.sh]: Will update.

Actually reason for the split was that Mac sed is not 100% compatible, `-i` takes additional arguments. (And I wanted to test locally)

**Comment by jacogr** [comment]
Ok, will try `--precice branch` locally. (Definately cleaner - should have done so already, @arkpar mentioned it as well)

**Comment by ngotchac** [comment]
Why not simply:

```bash
sed -i '/^parity-ui-precompile/ { s/dev/master/g; }' dapps/ui/Cargo.toml
```

and have in `dapps/ui/Cargo.toml`:

```
parity-ui-precompiled = { git = "https://github.com/ethcore/js-precompiled.git", optional = true, branch = "master" }
```

**Comment by jacogr** [comment]
Cannot do s/master/$BRANCH/ - on beta or stable it is not `master`. It may be when newly moved, but for e.g. stable it could be beta when just moved. (Actually this is why we had a bum 1.4.12 release)

We get this wrong all the time since we do it manually.

**Comment by ngotchac** [comment]
So 

```
sed -i '/^parity-ui-precompile/ { s/branch = ".*"/branch = "dev"/g; }' dapps/ui/Cargo.toml
```

?

File: js/scripts/release.sh [hunk 2]
```diff
@@ -89,11 +89,13 @@ fi
 
 echo "*** Updating cargo parity-ui-precompiled#$PRECOMPILED_HASH"
 git submodule update
+sed -i "/^parity-ui-precompiled/ { s/branch = \".*\"/branch = \"$BRANCH\"/g; }" dapps/ui/Cargo.toml
```
Comment (ngotchac) [inline js/scripts/release.sh]: Just out of curiosity : why `"` instead of `'` (which would permit having `branch = "$BRANCH"`) ?

**Comment by tomusdrw** [comment]
Can we also add a  warning comment in `dapps/ui/Cargo.toml` that it should always be a single line and that it's overwritten by build?

File: js/scripts/release.sh (hunk 2 above)
Comment (jacogr) [inline js/scripts/release.sh]: With ' the shell doesn't expand the variables, with " it does. (Hence the escape jiggle)

**Pull request merged**